src/runtime_src/core/common/api/xrt_kernel.cpp

```diff
@@ -1297,6 +1346,14 @@ class run_impl
     arg.set(arg_setter.get(), args);
   }
 
+  void
+  set_sk_args(const uint8_t* args, const uint32_t size)
+  {
+    //TODO This is allowed only for soft kernels
+    arg_setter->set_sk_args(args, size);
+  }
```

Comment:
Who will call this and how?  Soft kernels will not be a concept exposed at the API level so I am curious how this function will end being called?